astropy/modeling/parameters.py

```diff
@@ -728,22 +728,22 @@ def __rtruediv__(self, val):
         return val / self.value
 
     def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
+        return np.asanyarray(self) == np.asanyarray(val)
```

Comment:
Yes, if `__array__` can also return a `Quantity`, the first suggestion I gave will work fine.